File: threads/java/seq.java
public class seq {

    static void countdown(int n) {
    	while (n > 0) { n-=1; }
    }
	
    public static void main(String [] args) {
        final int COUNT = 80000000;
        long start = System.nanoTime();
	countdown(COUNT);
        long end = System.nanoTime();
        System.out.println("elapsed time = " + (end-start) + " nanoseconds");
    }
}

File: threads/c/seq.c
#include <stdio.h>
#include <stdint.h>
#include <stdlib.h>
#include <time.h>

#define BILLION 1000000000L

void countdown(int n) {
  while (n > 0) {
    n -= 1;
  }
}

main() {
  int COUNT = 80000000;

  uint64_t diff;
  struct timespec start, end;

  clock_gettime(CLOCK_MONOTONIC, &start);
  countdown(COUNT);
  clock_gettime(CLOCK_MONOTONIC, &end);

  diff = BILLION * (end.tv_sec - start.tv_sec) + end.tv_nsec - start.tv_nsec;
  printf("elapsed time = %llu nanoseconds\n", (long long unsigned int) diff);

  exit(0);
}
